src/control/run_go_tests.sh

```diff
@@ -117,50 +117,6 @@ function setup_environment()
 	fi
 }
 
-function emit_junit_failure()
-{
-    local cname="run_go_tests"
-    local tname="${1:-subtest}"
-    local fname="${DAOS_BASE}/test_results/${cname}.${tname}.xml"
-
-    local teststr="    <testcase classname=\"$cname\" name=\"$tname\">
-    <failure type=\"format\">
-      <![CDATA[$2
-        ]]>
-    </failure>
-    </testcase>"
-
-    cat > "${fname}" << EOF
-<?xml version="1.0" encoding="UTF-8" ?>
-<testsuites>
-  <testsuite tests="1" failures="1" errors="0" skipped="0" >
-    ${teststr}
-  </testsuite>
-</testsuites>
-EOF
-}
-
-function check_formatting()
```

Comment:
I think this function is still useful for developers who run the script locally (anyone who touches Go code is encouraged to run it prior to pushing a PR). ... After reading the rest of this PR, I see that you extracted it to a standalone script. I guess maybe we could also add that script to the pre-commit git hooks?

Reply:
The logic with this file was to move the check to where it ran outside of the Intel CI infrastructure so that the results were available to all.  This just seemed an easy thing to include and it doesn't natively support JUnit - which works with Jenkins but not with GitHub actions unfortunately.

I'm torn between wanting to run things quicker with the results available to all in GitHub actions or with the better reporting but taking longer and visible to fewer people.

It's easy enough to back out this change, I just saw it as something easy to move but hadn't considered your point about having a single entry point for developers - that's nice that you've got that process that works.

Reply:
I had completely forgotten about the gofmt githook that I added in December of 2022. I have no problem with removing this from the test runner script and into its own GHA-compatible script. At some point, it would be nice to make the GHA gofmt check work like the other linters, so that it shows the required updates inline as comments, but that could be done later.

Reply:
To make annotations work then you need the source line, gofmt gives you a diff but it's not easy to go from there to the line.

You've just reminded me of something though, in Jenkins checkpatch we used to have a process where we had scripts that could "fix" the source tree by patching them as gofmt has an option to do, then our code would run "git diff" to see where the changes were and if they were in a line modified by the PR then they'd annotate them as comments to the PR.  This was used by codespell and a gofmt hook, but was also really flaky and used to break a lot so I removed it when I moved codespell to GitHub actions.

Reply:
I deleted this part of the change.  The intention wasn't ever to change the workflow, I just wanted to be able to get the gofmt results viewable outside of Intel CI systems.